File: androidApp/src/main/java/com/blackopalsolutions/goneviral/android/model/response/GetGoalResponse.java
package com.blackopalsolutions.goneviral.android.model.response;

import com.blackopalsolutions.goneviral.android.model.domain.Goal;

public class GetGoalResponse extends Response {

    private final Goal goal;

    public GetGoalResponse(Goal goal) {
        super(goal != null);
        this.goal = goal;
    }

    public GetGoalResponse(String message) {
        super(message);
        this.goal = null;
    }

    public Goal getGoal() {
        return goal;
    }
}

File: shared/src/commonMain/kotlin/com/blackopalsolutions/goneviral/model/response/GetGoalResponse.kt
package com.blackopalsolutions.goneviral.model.response

import com.blackopalsolutions.goneviral.model.domain.Goal

class GetGoalResponse : Response {
    val goal: Goal?

    constructor(goal: Goal) : super(true) {
        this.goal = goal
    }

    constructor(message: String) : super(message) {
        goal = null
    }
}